Source/Dafny/Verifier/Translator.BoogieFactory.cs

```diff
@@ -184,7 +184,21 @@ enum BuiltinFunction {
       return null;
     }
 
-    Bpl.Expr RemoveLit(Bpl.Expr expr) {
+    static Bpl.Cmd CaptureState(Bpl.IToken tok, bool isEndToken, string/*?*/ additionalInfo) {
+      Contract.Requires(tok != null);
+      Contract.Ensures(Contract.Result<Bpl.Cmd>() != null);
+      var col = tok.col + (isEndToken ? tok.val.Length : 0);
+      string description = String.Format("{0}{1}", ErrorReporter.TokenToString(tok), additionalInfo == null ? "" : (": " + additionalInfo));
+      Bpl.QKeyValue kv = new Bpl.QKeyValue(tok, "captureState", new List<object>() { description }, null);
```

Comment:
I think it's easier to read if these variables are declared with just `var`, rather than explicitly giving their types (since the types are indeed the types of the RHSs of the assignments).